forta-bots/No-Settle/src/agent.spec.ts

```diff
@@ -0,0 +1,212 @@
+      blockHash: keccak256("fefsd"),
+    };
+
+    mockProvider.addFilteredLogs(filter0, []);
+    const findings: Finding[] = await handleBlock(blockEvent);
+
+    expect(findings).toStrictEqual([testCreateFinding(TEST_DAYS_THRESHOLD, "1423423422")]);
+  });
+
+  it("should fetch past logs correctly", async () => {
+    initialize = provideInitialize(mockNetworkManager as any, mockProvider as any, TEST_DAYS_THRESHOLD);
+    const blockNumber = 1234354353;
+    when(mockProvider.getBlockNumber).calledWith().mockReturnValue(blockNumber);
+    const filter = {
+      address: mockNetworkManager.TeleportJoin,
+      topics: [SETTLE_EVENT_TOPIC],
+      fromBlock: blockNumber - TEST_DAYS_THRESHOLD * 6100,
+      toBlock: blockNumber,
+    };
+    const logs = [
+      {
+        blockNumber: blockNumber - 1,
+        blockHash: keccak256("w43e23"),
+        transactionIndex: 2,
+        removed: false,
+        address: mockNetworkManager.TeleportJoin,
+        data: keccak256("dataData2"),
+        topics: [SETTLE_EVENT_TOPIC],
+        transactionHash: keccak256("tHash2"),
+        logIndex: 3,
+      },
+    ];
+    mockProvider.addFilteredLogs(filter, logs);
+    when(mockProvider.getBlock)
+      .calledWith(blockNumber - 1)
+      .mockReturnValue({ timestamp: 3242334 });
+    await initialize();
+    expect(mockProvider.getBlockNumber).toHaveBeenCalled();
+    expect(mockProvider.getNetwork).toHaveBeenCalled();
+    expect(mockProvider.getLogs).toHaveBeenCalled();
+    expect(mockProvider.getBlock).toHaveBeenCalledWith(blockNumber - 1);
+  });
+
+  it("should return a finding when the threshold has been exceeded", async () => {
+    handleBlock = provideHandleBlock(
+      mockNetworkManager as any,
+      mockProvider as any as ethers.providers.Provider,
+      TEST_DAYS_THRESHOLD
+    );
+
+    const blockEvent1: BlockEvent = new TestBlockEvent()
+      .setHash(keccak256("hash3"))
+      .setTimestamp(4464123422)
+      .setNumber(3456);
+
+    const filter = {
+      address: TEST_TELEPORT_JOIN,
+      topics: [SETTLE_EVENT_TOPIC],
+      blockHash: keccak256("hash3"),
+    };
+
+    const logs = [
+      {
+        blockNumber: 3456,
+        blockHash: blockEvent1.blockHash,
+        transactionIndex: 2,
+        removed: false,
+        address: mockNetworkManager.TeleportJoin,
+        data: keccak256("dataData2"),
+        topics: [SETTLE_EVENT_TOPIC],
+        transactionHash: keccak256("tHash3"),
+        logIndex: 3,
+      },
+    ];
+
+    const blockEvent2: BlockEvent = new TestBlockEvent()
+      .setHash(keccak256("hash5"))
+      .setTimestamp(5564123422) //exceeded
+      .setNumber(13456);
+
+    const filter2 = {
+      address: TEST_TELEPORT_JOIN,
+      topics: [SETTLE_EVENT_TOPIC],
+      blockHash: keccak256("hash5"),
+    };
+
+    mockProvider.addFilteredLogs(filter, logs).addFilteredLogs(filter2, []);
+
+    await handleBlock(blockEvent1);
+    const findings3 = await handleBlock(blockEvent2);
```

Comment:
please check the findings returned in L162, and change the name of L163 to `findings2` I guess